rm-datasource/src/main/java/io/seata/rm/datasource/DataSourceProxy.java

```diff
@@ -141,7 +142,35 @@ public String getResourceGroupId() {
     @Override
     public String getResourceId() {
         if (jdbcUrl.contains("?")) {
-            return jdbcUrl.substring(0, jdbcUrl.indexOf('?'));
```

Comment:
Keep the original clear, suggest:
return DBType.POSTGRESQL.name().equalsIgnoreCase(getDbType())? getPgResourceId() : jdbcUrl.substring(0,jdbcUrl.indexOf('?'));

Reply:
fixed.

Reply:
Only separate the different parts.
if (!jdbcUrl.contains("?")) is the same

Reply:
as you say, the code would be like this 
```
if (jdbcUrl.contains("?")) {
    if (DBType.POSTGRESQL.name().equalsIgnoreCase(getDbType())) {
        getPGResourceId();
    } else {
      do something....
    }
} else {
    do something
}
```
the code is less, but the logic is more complicated.